src/meta/meta_apply.c

```diff
@@ -10,55 +10,71 @@
 
 /*
  * __meta_btree_apply --
- *     Apply a function to all files listed in the metadata, apart from the metadata file.
+ *     Conditionally apply a function to the given file URI.
  */
 static inline int
-__meta_btree_apply(WT_SESSION_IMPL *session, WT_CURSOR *cursor,
+__wt_meta_btree_apply(WT_SESSION_IMPL *session, const char *uri,
   int (*file_func)(WT_SESSION_IMPL *, const char *[]),
   int (*name_func)(WT_SESSION_IMPL *, const char *, bool *), const char *cfg[])
 {
     WT_DECL_RET;
     int t_ret;
-    const char *uri;
     bool skip;
 
+    skip = false;
+    if (name_func != NULL)
+        ret = name_func(session, uri, &skip);
+
+    if (file_func == NULL || skip || !WT_PREFIX_MATCH(uri, "file:"))
+        return (ret);
+
     /*
-     * Accumulate errors but continue through to the end of the metadata.
+     * We need to pull the handle into the session handle cache and make sure it's referenced to
+     * stop other internal code dropping the handle (e.g in LSM when cleaning up obsolete chunks).
+     * Holding the schema lock isn't enough.
+     *
+     * Handles that are busy are skipped without the whole operation failing. This deals among other
+     * cases with checkpoint encountering handles that are locked (e.g., for bulk loads or verify
+     * operations).
      */
-    while ((t_ret = cursor->next(cursor)) == 0) {
-        if ((t_ret = cursor->get_key(cursor, &uri)) != 0 || strcmp(uri, WT_METAFILE_URI) == 0) {
-            WT_TRET(t_ret);
-            continue;
-        }
+    if ((t_ret = __wt_session_get_dhandle(session, uri, NULL, NULL, 0)) != 0) {
+        WT_TRET_BUSY_OK(t_ret);
+    }
 
-        skip = false;
-        if (name_func != NULL && (t_ret = name_func(session, uri, &skip)) != 0) {
-            WT_TRET(t_ret);
-            continue;
-        }
+    WT_SAVE_DHANDLE(session, WT_TRET(file_func(session, cfg)));
+    WT_TRET(__wt_session_release_dhandle(session));
 
-        if (file_func == NULL || skip || !WT_PREFIX_MATCH(uri, "file:"))
-            continue;
+    return (ret);
+}
+/*
+ * __meta_btree_apply --
+ *     Walk all files listed in the metadata, apart from the metadata file and LAS file. Applying a
+ *     given function to each file. At the end apply this function to the LAS file.
+ */
+static inline int
+__meta_btree_walk_and_apply(WT_SESSION_IMPL *session, WT_CURSOR *cursor,
```

Comment:
__meta_btree_apply_all is used when we are required to open files for checkpointing, which is different to the usual applying checkpoint to only opened files. As such I split out the functionality here and moved the LAS file to the end.